libeos-payg/manager.c

```diff
@@ -649,12 +650,12 @@ set_expiry_time (EpgManager *self,
  *
  * @now will typically be the value of (g_get_real_time() / G_USEC_PER_SEC).
  * It is parameterised to allow easy testing. */
-static void
+static gint64
 extend_expiry_time (EpgManager *self,
                     guint64     now_secs,
                     EpcPeriod   period)
 {
-  guint64 span_secs;
+  gint64 span_secs;
```

Comment:
Changing the type of this variable means the `span_secs = G_MAXUINT64` statement below is illegal, because `G_MAXUINT64` does not fit into a signed `gint64`.

Reply:
Ah, you are right - should I alter the type of the set_expiry_time function as well to keep it consistent across the code?

Reply:
I think you should either change the type of `set_expiry_time()`'s parameter to `gint64` and add `g_return_if_fail (span_secs >= 0);`  to the top; or leave this variable as `guint64` and add `g_return_val_if_fail (span_secs >= G_MAXINT64, G_MAXINT64)` at the bottom of this function. Personally I think I'd prefer the first option. It's really awkward that the AddCode method needs to return a signed value, but it really does have to be because the provider in eos-payg-nonfree can reduce the amount of credit in response to a call to AddCode.

Either way, you'll need to change the EPC_PERIOD_INFINITE branch of the switch.